CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.4)
+cmake_minimum_required(VERSION 3.9.1)
```

Comment:
this is pretty strict. 16.04 ubuntu provides only 3.5.1 or so... Do we definitely require the most recent cmake?

Reply:
At least under Windows. I have tried to get a successful Linux build, but it fails so far in any case.

Reply:
What features of cmake 3.9.1 do we need?

Reply:
Leaving the version as is for now, until the Linux build works and we know which cmake version we actually need there.